classes/StaffManager.php

```diff
@@ -513,60 +517,77 @@ public function canIssueOW($staff_level) {
     }
 
     public function banUser($ban_type, $ban_length, $user_data) {
-        if($ban_length > 0 || $ban_length == self::PERM_BAN_VALUE) {
-            $ban_data = json_decode($user_data['ban_data'], true);
-            $ban_expire = ($ban_length == self::PERM_BAN_VALUE ? $ban_length : time() + ($ban_length * 86400));
-            $old_ban_remaining = false;
-
-            //Create new ban data for storage
-            if($ban_data == null || empty($ban_data)) {
-                $ban_data = json_encode(array(
-                    $ban_type => $ban_expire
-                ));
+        $ban_data = json_decode($user_data['ban_data'], true);
+        $old_ban_remaining = false;
+
+        if($ban_length == self::PERM_BAN_VALUE) {
+            $ban_expire = $ban_length;
+        }
+        else {
+            // Bans are reported in minutes, add second value to current time
+            $ban_expire = time() + ($ban_length * 60);
+        }
+
+        //Create new ban data for storage
+        if($ban_data == null || empty($ban_data)) {
```

Comment:
I think these two are redundant, can probably just do
```suggestion
        if(empty($ban_data)) {
```